src/lib/php/Dao/HighlightDao.php

```diff
@@ -196,4 +196,24 @@ public function getHighlightEntries(ItemTreeBounds $itemTreeBounds, $licenseId =
     $highlightEntries = array_merge(array_merge($highlightDiffs,$highlightKeywords),$highlightBulk);
     return $highlightEntries;
   }
+
+  /**
+   * @param licenseMatchId
+   * @return page number
+   */
+  public function getPageNumberOfHighlightEntry($licenseMatchId)
+  {
+    $row = $this->dbManager->getSingleRow(
+      "SELECT (
+                (
+                  SELECT start FROM highlight WHERE fl_fk=$1 ORDER BY start DESC LIMIT 1
+                ) / (
+                  SELECT conf_value FROM sysconfig WHERE variablename LIKE 'BlockSizeText'
+                )::numeric
+              )::bigint
+       AS page;",
+      array($licenseMatchId)
+    );
+    return $row['page']-1;
```

Comment:
If the `page` is `0`, the function will return `-1`.

So, if there are no pages for the file, or the highlight is on the first page itself, the link generated will return a blank page.

Reply:
Or, if the license is on page 1, I am getting page 0.

Reply:
Yes, the pagination starts from 0 for licenses.

Reply:
For some reason it started with 1 on my Ubuntu Xenial machine and with 0 on Debian Buster.

I will investigate and let you know.

Reply:
I just recalled what I reported first.
To rephrase,

If there are no page on the file (file is loaded at a one go), the `page` returned is `-1` which is invalid.

So maybe a check `<0` will be good.